jetty-http/src/main/java/org/eclipse/jetty/http/HttpURI.java

```diff
@@ -486,45 +542,37 @@ else if (c == '/')
             case SCHEME_OR_PATH:
                 _path = uri.substring(mark, end);
                 break;
-
             case HOST_OR_PATH:
                 _path = uri.substring(mark, end);
                 break;
-
             case HOST:
                 if (end > mark)
                     _host = uri.substring(mark, end);
                 break;
-
             case IPV6:
                 throw new IllegalArgumentException("No closing ']' for ipv6 in " + uri);
-
             case PORT:
                 _port = TypeUtil.parseInt(uri, mark, end - mark, 10);
                 break;
-
             case ASTERISK:
                 break;
-
             case FRAGMENT:
                 _fragment = uri.substring(mark, end);
                 break;
-
             case PARAM:
                 _path = uri.substring(pathMark, end);
                 _param = uri.substring(mark, end);
                 break;
-
             case PATH:
+                checkSegment(uri, segment, end, false);
                 _path = uri.substring(pathMark, end);
                 break;
-
             case QUERY:
                 _query = uri.substring(mark, end);
                 break;
         }
```

Comment:
Please add `default` with explicit `break` if that's the case, or throwing if it should not be reached.